Atomic.md

```diff
@@ -0,0 +1,618 @@
+
+While this field seems complicated/confusing to work with, it can easily be abstracted away (e.g. as a part of autofilling) in tooling, and it's easy for `rippled` to check a hash doesn't match its corresponding transaction in `RawTransaction`.
+
+### 2.5. `AtomicSigners`
+
+This field operates similarly to [multisign](https://xrpl.org/docs/concepts/accounts/multi-signing/) on the XRPL. It is only needed if multiple accounts' transactions are included in the `Atomic` transaction; otherwise, the normal transaction signature provides the same security guarantees.
+
+|FieldName | Required? | JSON Type | Internal Type |
+|:---------|:-----------|:---------------|:------------|
+|`Account`|✔️|`string`|`STAccount`|
+|`SigningPubKey`| |`string`|`STBlob`|
+|`Signature`| |`string`|`STBlob`|
+|`Signers`| |`array`|`STArray`|
+
+#### 2.5.1. `Account`
+
+This is an account that has at least one inner transaction.
+
+#### 2.5.2. `SigningPubKey` and `Signature`
+
+These fields are included if the account is signing with a single signature (as opposed to multi-sign). They sign the `Flags` and `TxnIDs` fields.
+
+#### 2.5.3. `Signers`
+
+This field is included if the account is signing with multi-sign (as opposed to a single signature). It operates equivalently to the [`Signers` field](https://xrpl.org/docs/references/protocol/transactions/common-fields/#signers-field) used in standard transaction multi-sign. This field holds the signatures for the `Flags` and `TxnIDs` fields.
+
+### 2.6. Metadata
+
+The inner transactions will be committed separately to the ledger and will therefore have separate metadata. This is to ensure better backwards compatibility for legacy systems, so they can support `Atomic` transactions without needing any changes to their systems.
+
+For example, a ledger that only has one `Atomic` transaction containing 2 inner transactions would look like this:
+```
+[
+  OuterTransaction,
+  InnerTransaction1,
+  InnerTransaction2
+]
+```
+
+#### 2.6.1. Outer Transactions
+
+Each outer transaction will only contain the metadata for its sequence and fee processing, not for the inner transaction processing.
+
+There will also be a list of which transactions were actually processed, which is useful for the `ONLYONE` and `BATCH` atomicity types, since those may only process a subset of transactions, and for debugging with all atomicity types. This section will be called `AtomicExecutions`.
+
+It will contain a list of objects that have the following fields for every transaction that is processed (successfully or unsuccessfully):
+
+|FieldName | Required? | JSON Type | Internal Type |
+|:---------|:-----------|:---------------|:------------|
+|`TransactionHash`|✔️|`string`|`STUInt256`|
+|`TransactionResult`|✔️|`string`|`STUInt8`|
```

Comment:
Would you rather it be stored as a `VL` string name then?

Reply:
I don't know... The UInt8 saves a lot of space.... What do you think about two fields: one UInt8 for `tec` and `tes`, and one string for everything else?

Reply:
That's a bit complicated to work with...
What about a UInt16 that's the TER code value + 400? Or something like that, where there's a transformation?

Reply:
> That's a bit complicated to work with and feels a bit confusing... 

Yeah, I can see that.

> What about a UInt16 that's the TER code value + 400? Or something like that, where there's a transformation?

The real problem is not that the values are negative, but that they're intentionally unstable. Otherwise the transformation is a great idea.

Reply:
I just looked at `TER.h` and noticed that there's a notation that the values are used in the `ripple-binary-codec`, so I guess that ship has already sailed. Still, please note it in the spec, maybe the FAQ.